equinox/nn/linear.py

```diff
@@ -57,6 +57,15 @@ def __call__(
         - `key`: Ignored; provided for compatibility with the rest of the Equinox API.
             (Keyword only argument.)
 
+        !!! info
+
+        Note that in contrast to [PyTorch](https://pytorch.org/docs/stable/generated/torch.nn.Linear.html) or other
+        deep learning frameworks where the input `x` is allowed to be a higher order tensor of shape
+        `(in_features, ...)`, here we require `x` to be a vector. If you want to use higher order tensors as
```

Comment:
**(A)** PyTorch actually has `(..., in_features)`.
This means the Equinox example would equivalently change to
```python
linear = eqx.nn.Linear(...)
jax.vmap(linear)(x)
```

**(B)** The region below `!!! info` should be indented four spaces in order to format correctly in the documentation.

If you want to look at the documentation locally, by the way:
```
bash> pip install -r docs/requirements.txt
bash> mkdocs serve
bash> mkdocs serve  # again
```
and then navigate to `localhost:8000` in a web browser.

**(C)** This isn't actually unique to the `eqx.nn.Linear` layer. All Equinox operations operate on a single element, and need a `jax.vmap` to work over a batch. Perhaps it would be worth adding this to the FAQ? (Either as well as or instead of, whichever you think best.)

Reply:
Hi, thanks for the comments. 
(A) good catch, that is potentially confusing, and I will change it accordingly.
(B) oops, I checked and actually didn't realize it didn't look the way it should. Will fix that.
(C) I guess this would make sense. I'm pretty sure you will have many users like me coming to equinox as users of diffrax who don't have a strong background in JAX.
So, summing up I'll make the suggested edits and update the PR.

Reply:
I have added the fixes described above. Note that I made a small change to the pre-commit hooks as well to address https://github.com/psf/black/issues/2964.

Reply:
Thanks! This basically LGTM; I'll handle merging this in a bit.
Regarding click -- lower down the thread you link, it looks like updating to Black 22.3.0 should fix things, instead of pinning click?

Reply:
Yeah, I was a bit hesitant about this, though, since the formatting rules might have changed in between versions. I've had this problem a couple of times in the past, causing reformatting of the entire codebase. But probably just worth a try...